tests/integration/security/rbac/rbac_v2_test.go

```diff
@@ -120,13 +128,49 @@ func TestRBACV2(t *testing.T) {
 	// Sleep 5 seconds for the policy to take effect.
 	// TODO(pitlv2109: Check to make sure policies have been created instead.
 	time.Sleep(5 * time.Second)
-	for _, conn := range connections {
+	for _, tc := range cases {
 		retry.UntilSuccessOrFail(t, func() error {
-			return connection.CheckConnection(t, conn)
+			return checkRBACRequest(tc)
 		}, retry.Delay(time.Second), retry.Timeout(10*time.Second))
 	}
 }
 
+// checkRBACRequest checks if a request is successful under RBAC policies.
+// Under RBAC policies, a request is consider successful if:
+// * If the policy is deny:
+// *** For HTTP: response code is 403
+// *** For TCP: EOF error
+// * If the policy is allow:
+// *** Response code is 200
+func checkRBACRequest(tc testCase) error {
+	req := tc.request
+	ep := req.To.EndpointForPort(req.Port)
+	if ep == nil {
+		return fmt.Errorf("cannot get upstream endpoint for connection test %v", req)
+	}
+	var expectedRespCode string
+	if tc.isRequestAllowed {
+		expectedRespCode = connection.AllowHTTPRespCode
+	} else {
+		expectedRespCode = connection.DenyHTTPRespCode
+	}
+	resp, err := req.From.Call(ep, apps.AppCallOptions{Protocol: req.Protocol, Path: req.Path})
+	if err != nil {
+		if req.Port == connection.TCPPort && strings.Contains(err.Error(), "EOF") {
+			return nil
```

Comment:
This is testing TCP/Deny cases, correct? Do we need to check the expectedRespCode or see if tc.isRequestAllowed is false?

Reply:
Good catch. This is testing TCP deny cases. It need to check if tc.isRequestAllowed is false as well.  Will update. Thanks.

Reply:
This should be merged with the `isRequestAllowed` and handled together like the HTTP cases.
```
if req.isRequestAllowed:
  if req.Port == connection.TCPPort
    # Check TCP allowed, otherwise test fail
  else
    # Check HTTP allowed, otherwise test fail
else:
  if req.Port == connection.TCPPort
     # Check TCP denied, otherwise test fail
  else
     # Check HTTP denied, otherwise test fail
```

Reply:
I just updated the code.

As for Yangmin's comment, this looks cleaner, but there are two things that the current approach* offers:
* If `Call` returns some error, we can catch it early and check if this error is expected (when protocol is tcp and the policy is deny) or not (and return early)
* We won't have to repeat `expectedRespCode == resp[0].Code` in both http&tcp allow and http deny. 

Let me know which approach you prefer. The current one's approach:
If `err != nil`, we check if the err is expected. If it is, check if tcp deny. If not, return error.
Since we got some response, check if `expectedRespCode == resp[0].Code` and return success.
Finally, if nothing passes, return with error.

Reply:
Let's go with the cleaner approach. This is test and the following doesn't seem any repetitive to me. (renamed `isRequestAllowed` to `expectAllowed`)
```
if req.expectAllowed:
  if resp[0].Code != "200":
    t.Error("want 200, got ...")
else:
  if req.Port == connection.TCPPort
     if !strings.Contains(err.Error(), "EOF"):
       t.Error("want EOF, got ...")
  else
     if !(len(resp) > 0 && "403" == resp[0].Code):
       t.Error("want 403, got ...")
```

Reply:
Updated. Thanks. I added a check to return early if we got an unexpected error.